src/packet-ffxiv.c

```diff
@@ -145,8 +150,15 @@ static int dissect_frame(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree, vo
     next_tvb = tvb_new_subset_remaining(tvb, FRAME_HEADER_LEN);
   }
 
-  tcp_dissect_pdus(next_tvb, pinfo, frame_tree, TRUE, BLOCK_HEADER_LEN,
-    get_message_length, dissect_message, data);
+  offset = 0;
```

Comment:
Offset is the offset in the current TVB and at this point should be set to the start of the payload data. Resetting it to zero would include the frame header in the message_tvb. I would probably also rename `next_tvb` to `payload_tvb` just so it's clearer what it gets used for.